### DISCUSSION TITLE
Async-SQS slower than AWS SDK?

### DISCUSSION THREAD

**Discussion by emajo** [async-aws/aws#1383] [Q&A]
Hi, I run a few tests with the async-sqs library and the official aws sdk, I was expecting the async library to be much faster but sending a message I get an average of 65ms using the async lib (without using the result) and 55ms using the official lib (sync, getting the result), the code is the same, only the SQSClient import is different:

```
  $client = new SqsClient([
      'region' => env('AWS_DEFAULT_REGION'),
  ]);

  $params = [
      'QueueUrl' => env('QUEUE_URL'),
      'MessageBody' => 'sqs benchmark',
  ];

  $tStart = microtime(true);

  $client->sendMessage($params);

  $time = microtime(true) - $tStart;
```
If someone knows if I'm doing something wrong I would appreciate a lot.

**Comment by GrahamCampbell** [comment]
> I was expecting the async library to be much faster but sending a message I get an average of 65ms using the async lib

Why were you expecting this? They both just go and use curl to hit the server, behind the scenes. It is largely no different, especially with the super small response bodies from sqs.

The difference could be down to the async aws having more files to load for the first time compared to the official sdk. Are you able to compare performance of the 2nd request, reusing the same client instance?

**Comment by emajo** [comment]
Before digging deeper into the code I was thinking that if I didn't use the result the message would have been sent as 'fire and forget', so I was surprised about the benchmarks.

Since my use case was to send a message for every http request made to my backend ∼50ms were too much (I don't have the possibility to reuse the same connection), so I moved away from SQS and therefore I cannot do further benchmarks on it.
